prac_07/project_manager.py

```diff
@@ -0,0 +1,150 @@
+"""
+estimate: 3 hrs
+actual: 4hrs
+"""
+
+from project import Project
+import datetime
+
+FILENAME = "projects.txt"
+MENU = "- (L)oad projects \n- (S)ave projects \n- (D)isplay projects \n- (F)ilter projects by date \n- (A)dd new project " \
+       "\n- (U)pdate project \n- (Q)uit"
+projects = []
+
+
+def main():
+    """Program to load and update projects"""
+    load_projects()
+    print(MENU)
+    choice = input(">>> ").lower()
+    while choice != "q":
+        if choice == "l":
+            load_new_projects()
+        elif choice == "d":
+            display_projects()
+        elif choice == "a":
+            add_new_project()
+        elif choice == "s":
+            save_projects()
+        elif choice == "f":
+            display_projects_by_date()
+        elif choice == "u":
+            update_project_details()
+        else:
+            print("Invalid menu choice")
+        print(MENU)
+        choice = input(">>> ").lower()
+    print("Thank you for using custom-built project management software.")
+    save_ending_projects()
+
+
+def save_ending_projects():
+    """Saves the projects"""
+    with open(FILENAME, 'w') as out_file:
+        for project in projects:
+            print(project, file=out_file, end="\n")
+
+
+def update_project_details():
+    """Updates a projects completion and priority"""
+    for i, project in enumerate(projects):
+        print(f"{i} {project}")
+    project_choice = int(input("Project choice: "))
+    print(projects[project_choice])
+    percentage = int(input("New percentage: "))
+    priority = int(input("Priority: "))
+    projects[project_choice] = Project(projects[project_choice].name, projects[project_choice].start_date, priority,
+                                       projects[project_choice].cost_estimate, percentage)
+
+
+def display_projects_by_date():
+    """Displays the projects after a entered date"""
+    date_string = input("Show projects that start after date (dd/mm/yyyy): ")  # e.g., "30/9/2022"
+    date_string = datetime.datetime.strptime(date_string, "%d/%m/%Y").date()
```

Comment:
This variable name is slightly misleading, as it redefined from a string to a datetime object. Perhaps redefining it to different name after line 62 would be more readable.